src/components/templates/address-form.tsx

```diff
@@ -22,149 +22,156 @@ export type AddressPayload = {
-          label="Address 1"
           required={required}
           errors={errors}
         />
         <Input
-          {...form.register(path("address_2"), {
-            pattern: FormValidator.whiteSpaceRule("Address 2"),
+          {...form.register(path("province"), {
+            pattern: FormValidator.whiteSpaceRule("Province"),
           })}
-          placeholder="Address 2"
-          label="Address 2"
+          placeholder="Province"
+          label="Province"
           errors={errors}
         />
-        <div className="grid grid-cols-[144px_1fr] gap-large">
-          <Input
-            {...form.register(path("postal_code"), {
-              required: required
-                ? FormValidator.required("Postal code")
-                : false,
-              pattern: FormValidator.whiteSpaceRule("Postal code"),
-            })}
-            placeholder="Postal code"
-            label="Postal code"
-            required={required}
-            autoComplete="off"
-            errors={errors}
-          />
-          <Input
-            placeholder="City"
-            label="City"
-            {...form.register(path("city"), {
-              required: required ? FormValidator.required("City") : false,
-              pattern: FormValidator.whiteSpaceRule("City"),
-            })}
-            required={required}
-            errors={errors}
-          />
-        </div>
-        <div className="grid grid-cols-2 gap-large">
-          <Input
-            {...form.register(path("province"), {
-              pattern: FormValidator.whiteSpaceRule("Province"),
-            })}
-            placeholder="Province"
-            label="Province"
-            errors={errors}
-          />
-          <Controller
-            control={control}
-            name={path("country_code")}
-            rules={{
-              required: required ? FormValidator.required("Country") : false,
-            }}
-            render={({ field: { value, onChange } }) => {
-              return (
-                <NextSelect
-                  label="Country"
-                  required={required}
-                  value={value}
-                  options={countryOptions}
-                  onChange={onChange}
-                  name={path("country_code")}
-                  errors={errors}
-                />
-              )
-            }}
-          />
-        </div>
+        <Controller
+          control={control}
+          name={path("country_code")}
+          rules={{
+            required: required ? FormValidator.required("Country") : false,
+          }}
+          render={({ field: { value, onChange } }) => {
+            return (
+              <NextSelect
+                label="Country"
+                required={required}
+                value={value}
+                options={countryOptions}
+                onChange={onChange}
+                name={path("country_code")}
+                errors={errors}
+              />
```

Comment:
nit: Since this field can be optional it might make sense to add an `isClearable` to it so people can reset the field.